uploader/push_to_sling.py

```diff
@@ -36,74 +68,114 @@
+            continue
+        logger.debug('root: %s', root)
+        logger.debug('file: %s', file)
+        path = PurePath(root + '/' + file)
+
+        # These distinctions aren't important right now but they set us up for later
+        isTitle = matches(path, titleGlobs, 'title')
+        isModule = matches(path, moduleGlobs, 'module') if not isTitle else False
+        isResource = matches(path, resourceGlobs, 'resource') if not isModule else False
+
+        if isTitle or isModule or isResource:
+            base_name = path.stem
+
+            ppath = path
+            hiddenFolder = False
+            while not ppath == PurePath(args.directory):
+                logger.debug('ppath: %s', str(ppath.stem))
+                if ppath.stem[0] == '.':
+                    hiddenFolder = True
+                    break
+                ppath = ppath.parent
+            if hiddenFolder:
+                logger.info('Skipping %s because it is hidden.', str(path))
+                logger.info('')
+                continue
+
+            # parent directory
+            parent_dir_str = str(path.parent.relative_to(args.directory))
+            if parent_dir_str == '.':
+                parent_dir_str = ''
+            logger.debug('parent_dir_str: %s', parent_dir_str)
+            # file becomes a/file/name (no extension)
+
+            url = server + "/content/repositories/" + repository
+            if parent_dir_str:
+                url += '/' + parent_dir_str
+
+            logger.debug('base name: %s', base_name)
+
+            # Asciidoc content (treat as a module)
+            if path.suffix == '.adoc' or path.suffix == '.asciidoc':
+                print(path)
+                url += '/' + path.name
+                logger.debug(url)
+                data = {"jcr:primaryType": 'pant:module',
+                        "jcr:title": base_name,
+                        "jcr:description": base_name,
+                        "sling:resourceType": 'pantheon/modules',
+                        "pant:originalName": path.name,
+                        "asciidoc@TypeHint": 'nt:file'}
+                files = {'asciidoc': ('asciidoc', open(path, 'rb'), 'text/x-asciidoc')}
+
+                # Minor question: which is correct, text/asciidoc or text/x-asciidoc?
+                # It is text/x-asciidoc. Here's why:
+                # https://tools.ietf.org/html/rfc2045#section-6.3
+                # Paraphrased: "If it's not an IANA standard, use the 'x-' prefix.
+                # Here's the list of standards; text/asciidoc isn't in it.
+                # https://www.iana.org/assignments/media-types/media-types.xhtml#text
+
+                if not args.dry:
+                    r = requests.post(url, headers=HEADERS, data=data, files=files, auth=(args.user, pw))
+                    print(r.status_code, r.reason)
+                logger.debug('')
+            # Otherwise just upload as a regular file
+            else:
```

Comment:
Similarly, this should be:
```python
elif isResource:
```
We can deal with titles later.

Reply:
Yeah, I'm thinking that for the time being, rather than keying off of `isXxxx` flags, we keep the filename convention - BUT we use the `isXxxx` flag to set some new property on the .adoc uploads, like `pant:renderable` or something to that effect. We set it to true for `isModule` (and `isTitle`, hopefully?), and false for `isResource`.

That way, we still have asciidoc content where we expect it for JCR nodes regardless of whether it's a module/title/resource, but we're also not trying to render content that isn't standalone.

Reply:
We should discuss this one. I might see the value of the `renderable` field, but at this point we know the file naming convention is a no-go, so why not remove it?

Reply:
Now that I thought about it for a few minutes, I don't like the `pant:renderable` suggestion. We already have the `sling:resourceType` of `pantheon/modules`, now it probably makes sense to introduce `pantheon/titles` and `pantheon/resources` as well, and start keying off that.

Reply:
Remember there is sling's resource type, which is used for rendering; and JCRs node type, which is used more for hard node types. In this scenario, we should be referring to node types. So `pant:module` and `pant:title`.

Reply:
Ok, good thinking, we can use `jcr:nodeType` for this. Agreed.